app/src/main/java/com/globant/marvelmvvm/util/AllCharactersRecyclerViewAdapter.kt

```diff
@@ -16,12 +16,12 @@ class AllCharactersRecyclerViewAdapter(private val onItemClick: (String) -> Unit
     private var characterList: List<Character> = ArrayList()
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
-        return WeatherViewHolder(
+        return AllCharactersViewHolder(
             LayoutInflater.from(parent.context)
                 .inflate(R.layout.card_view_character_item_layout, parent, false), onItemClick)}
 
     override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
-        (holder as WeatherViewHolder).bind(this.characterList[position])
+        (holder as AllCharactersViewHolder).bind(this.characterList[position])
```

Comment:
is this cast necessary?

Reply:
It is not necessary, thanks by the explanation. Refactored!